src/Compilers/Shared/ShadowCopyAnalyzerAssemblyLoader.cs

```diff
@@ -11,7 +9,12 @@
 
 namespace Microsoft.CodeAnalysis
 {
-    internal sealed class ShadowCopyAnalyzerAssemblyLoader : DesktopAnalyzerAssemblyLoader
+    internal sealed class ShadowCopyAnalyzerAssemblyLoader :
+#if NET46
+        DesktopAnalyzerAssemblyLoader
+#else
+        CoreClrAnalyzerAssemblyLoader
+#endif
```

Comment:
This is... uh... not the greatest 😅

I think we should probably have a separate class of `AssemblyLoaderImplementation` that has Desktop vs. CoreClr, and then wrap that (as a field) in `AnalyzerAssemblyLoader`, that `ShadowCopyAnalyzerAssemblyLoader` inherits from. I wanted feedback before doing so, though.

(Important to note is that "shadow copy" here doesn't refer to the CLR's shadow copying, but rather we're literally manually copying the DLL someplace else, in a portable fashion)